test/gtest/ucs/test_sock.cc

```diff
@@ -53,6 +69,50 @@ class test_socket : public ucs::test {
             EXPECT_EQ(exp_str, test_str);
         }
     }
+
+    void verify_subnet_match(sa_family_t af, const char *ip_addr1,
+                             const char *ip_addr2,
+                             unsigned longest_common_prefix) const
+    {
+        struct sockaddr_storage storage1, storage2;
+        struct sockaddr *saddr1;
+        struct sockaddr *saddr2;
+        int matched;
+
+        saddr1 = (struct sockaddr*)&storage1;
+        saddr2 = (struct sockaddr*)&storage2;
+        fill_sockaddr(saddr1, af, ip_addr1, 1234);
+        fill_sockaddr(saddr2, af, ip_addr2, 1234);
+
+        size_t addr_size_bits = get_addr_size(saddr1) * CHAR_BIT;
+
+        for (size_t prefix = 0; prefix <= addr_size_bits; ++prefix) {
+            ASSERT_UCS_OK(ucs_sockaddr_is_same_subnet(saddr1, saddr2, prefix,
+                                                      &matched));
+            EXPECT_EQ(prefix <= longest_common_prefix, matched)
+                    << "failed to match subnet: ip1=" << ip_addr1
+                    << " ip2=" << ip_addr2 << " prefix=" << prefix
+                    << " max_common=" << longest_common_prefix
+                    << " (matched=" << matched << ")";
+        }
+    }
+
+    void verify_subnet_error(sa_family_t af1, sa_family_t af2, unsigned prefix,
+                             ucs_status_t expected_status) const
+    {
+        struct sockaddr_storage storage1, storage2;
+        struct sockaddr *sa1 = (struct sockaddr*)&storage1;
+        struct sockaddr *sa2 = (struct sockaddr*)&storage2;
```

Comment:
seems the addresses are not initialized

Reply:
what do you mean? they are initialized with the address of the stack vars

Reply:
I mean the contents of storage1, 2

Reply:
they are not required to be, as this test checks AF errors